From 61ffc5ec5ad1755d5d22b4386f8ad027bb59e1d5 Mon Sep 17 00:00:00 2001
From: Scott Murray <scott.murray@konsulko.com>
Date: Tue, 5 Oct 2021 12:10:23 -0400
Subject: [PATCH] rcar_gen3: plat: Do not panic on unrecognized boards

Replace the panic in bl2_populate_compatible_string with just putting
"unknown" into the compatible string for the FDT.  This allows us to
boot on the AGL reference hardware where the board type seems
unavailable.

Upstream-Status: Inappropriate [board specific usecase]

Signed-off-by: Scott Murray <scott.murray@konsulko.com>
---
 plat/renesas/rcar/bl2_plat_setup.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/plat/renesas/rcar/bl2_plat_setup.c b/plat/renesas/rcar/bl2_plat_setup.c
index bf077c483..7fcc75a13 100644
--- a/plat/renesas/rcar/bl2_plat_setup.c
+++ b/plat/renesas/rcar/bl2_plat_setup.c
@@ -489,8 +489,9 @@ static void bl2_populate_compatible_string(void *dt)
 					 "renesas,draak");
 		break;
 	default:
-		NOTICE("BL2: Cannot set compatible string, board unsupported\n");
-		panic();
+		ret = fdt_setprop_string(dt, 0, "compatible",
+					 "renesas,unknown");
+		break;
 	}
 
 	if (ret < 0) {
-- 
2.31.1

